src/core/config/remote/remote_config_manager.ts

```diff
@@ -0,0 +1,288 @@
+        }
+
+        const currentCommand = argv._.join(' ')
+        self.remoteConfig!.addCommandToHistory(currentCommand)
+
+        await self.save()
+      }
+    }
+  }
+
+  /**
+   * Builds a task for creating a new remote configuration, intended for use with Listr task management.
+   * Merges cluster mappings from the provided context into the local configuration, then creates the remote config.
+   *
+   * @returns a Listr task which creates the remote configuration.
+   */
+  public buildCreateTask (): ListrTask<ListrContext> {
+    const self = this
+
+    return {
+      title: 'Create remote config',
+      task: async (_, task): Promise<void> => {
+        const localConfigExists = this.localConfig.configFileExists()
+        if (!localConfigExists) {
+          throw new SoloError('Local config doesn\'t exist')
+        }
+
+        if (await self.load()) {
+          task.title = `${task.title} - ${chalk.red('Remote config already exists')}}`
+
+          throw new SoloError('Remote config already exists')
+        }
+
+        await self.create()
+      }
+    }
+  }
+
+  /* ---------- Utilities ---------- */
+
+  /**
+   * Retrieves the ConfigMap containing the remote configuration from the Kubernetes cluster.
+   *
+   * @returns the remote configuration data.
+   * @throws {@link SoloError} if the ConfigMap could not be read and the error is not a 404 status.
+   */
+  private async getConfigMap (): Promise<k8s.V1ConfigMap> {
+    try {
+      return await this.k8.getNamespacedConfigMap(constants.SOLO_REMOTE_CONFIGMAP_NAME)
+    } catch (error: any) {
+      if (error.meta.statusCode !== 404) {
+        throw new SoloError('Failed to read remote config from cluster', error)
+      }
+
+      return null
+    }
+  }
+
+  /**
+   * Creates a new ConfigMap entry in the Kubernetes cluster with the remote configuration data.
+   */
+  private async createConfigMap (): Promise<void> {
+    await this.k8.createNamespacedConfigMap(
+      constants.SOLO_REMOTE_CONFIGMAP_NAME,
+      constants.SOLO_REMOTE_CONFIGMAP_LABELS,
+      { 'remote-config-data': yaml.stringify(this.remoteConfig.toObject()) }
+    )
+  }
+
+  /** Replaces an existing ConfigMap in the Kubernetes cluster with the current remote configuration data. */
+  private async replaceConfigMap (): Promise<void> {
+    await this.k8.replaceNamespacedConfigMap(
+      constants.SOLO_REMOTE_CONFIGMAP_NAME,
+      constants.SOLO_REMOTE_CONFIGMAP_LABELS,
+      { 'remote-config-data': yaml.stringify(this.remoteConfig.toObject() as any) }
+    )
+  }
+
+  private setDefaultNamespaceIfNotSet (): void {
+    if (this.configManager.hasFlag(flags.namespace)) return
+
+    if (!this.localConfig?.currentDeploymentName) {
+      this.logger.error('Current deployment name is not set in local config', this.localConfig)
+      throw new SoloError('Current deployment name is not set in local config')
+    }
+
+    // TODO: Current quick fix for commands where namespace is not passed
```

Comment:
Which commands don't have namespace and are loading remoteConfig?

Reply:
node logs, and etc. Since we need the remote config loaded to update the lastExecuted command and command History

Reply:
I think it is okay if we don't capture history for non-mutating commands.  the history was an optional non-MVP item anyway.